### PR TITLE
Gameplay: Room transition & Teleport

### PR DESCRIPTION
**Pull request by Bernatmago**
Script that turns screen black, disables skybox and teleports u inside a room. Then u can go out of the room and the same transition happens the other way arround.

Prefab & Test scene included, when adding a new teleporter u only need to assign the player to it and move the teleporters to the correct position 👍

### CODE REVIEW

**Review by bernatgb — APPROVED** [review]
Working perfectly!

**Review by monikamnc — APPROVED** [review]
I saw some meta files created in this pr that are in other pr's also, don't know if it will generate conflicts (?)
Thanks for your effort!

File: Gameplay/RoomTeleporter.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
+#include "scriptingUtil/gameplaypch.h"
+#include "RoomTeleporter.h"
+
+// NOTE FOR BERNAT: To scripts be able to work, you need to define it's
```
Comment (Erick9Thor) [inline Gameplay/RoomTeleporter.cpp:4]: mmm hahaha

File: Gameplay/RoomTeleporter.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+void Hachiko::Scripting::RoomTeleporter::OnUpdate()
+{
+	if (!_player || !_outdoor_portal || !_room_portal || !_fade_image)
+	{
+		// If some reference is working the system cant work
+		return;
+	}
+	
+	if(Input::IsKeyDown(Input::KeyCode::KEY_T))
```
Comment (Erick9Thor) [inline Gameplay/RoomTeleporter.cpp:54]: This is for debug proposes?

Comment (Erick9Thor) [inline Gameplay/RoomTeleporter.cpp:54]: The key T it's for the interaction with the player, for trigger the scene change?

File: Gameplay/RoomTeleporter.cpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		_step_progress += Time::DeltaTime() / _fade_duration;
+		_fade_image->SetColor(float4::Lerp(_opaque_color, _clear_color, _step_progress));
+	}
+
+	// Step transition logic
+	if (_step_progress >= 1.f)
+	{
+		_step_progress = 0.f;
+		
+		if (_step == Step::FADE_IN)
```
Comment (Erick9Thor) [inline Gameplay/RoomTeleporter.cpp:132]: maybe it looks better with a switch, than if else if else... i know that it's the same, is just my opinion

File: Gameplay/RoomTeleporter.cpp [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	}
+	
+}
+
+void Hachiko::Scripting::RoomTeleporter::TeleportPlayer(bool indoors)
+{
+	if (indoors)
+	{
+		_player->GetTransform()->SetGlobalPosition(_room_portal->GetTransform()->GetGlobalPosition());
+		HE_LOG("Teleport indoors");
```
Comment (Erick9Thor) [inline Gameplay/RoomTeleporter.cpp:157]: This kind of logs works on gameplay¿?!

**Comment by Bernatmago** [comment]
> I saw some meta files created in this pr that are in other pr's also, don't know if it will generate conflicts (?) Thanks for your effort!

Merged dev and now all aligned 👍

**Pull request merged**